bot/exts/moderation/voice_gate.py

```diff
@@ -254,6 +254,10 @@ async def on_voice_state_update(self, member: Member, before: VoiceState, after:
             log.trace("User not in a voice channel. Ignore.")
             return
 
+        if isinstance(after.channel, discord.StageChannel):
+            log.trace("User joined a stage channel. Ignore.")
+            return
```

Comment:
would it be interesting to refactor like this?
```
not_applicable = any(
    member.bot,  # is a bot
    member.voice is None,  # not in a voice channel
    isinstance(after.channel, discord.StageChannel)  # user is muted in a StageChannel
)

if not_applicable:
    return
```
Or do you need the log.trace for each case?

Reply:
I found the separate trace logs for each case very useful during dev testing. I'm hesitant to make this change for that reason.

Reply:
ok makes sense indeed